services/web/server/src/simcore_service_webserver/projects/db.py

```diff
@@ -390,28 +410,40 @@ async def list_projects(  # pylint: disable=too-many-arguments
                         if not include_hidden
                         else sa.text("")
                     )
-                    & (
-                        (projects.c.prj_owner == user_id)
-                        | sa.text(
-                            f"jsonb_exists_any(access_rights_subquery.access_rights, {assemble_array_groups(user_groups)})"
-                        )
-                    )
                     & (
                         (projects_to_products.c.product_name == product_name)
                         # This was added for backward compatibility, including old projects not in the projects_to_products table.
                         | (projects_to_products.c.product_name.is_(None))
                     )
+                    & (
+                        projects_to_folders.c.folder_id == folder_id
+                        if folder_id
+                        else projects_to_folders.c.folder_id.is_(None)
+                    )
+                    & (
+                        projects.c.workspace_id == workspace_id  # <-- Shared workspace
+                        if workspace_id
+                        else projects.c.workspace_id.is_(None)  # <-- Personal workspace
+                    )
                 )
             )
-            if settings.WEBSERVER_FOLDERS:
+
+            if workspace_id is None and personal_workspace_user_id_or_none:
+                # If Personal workspace we check to which projects user has access
+                user_groups: list[RowProxy] = await self._list_user_groups(
+                    conn, personal_workspace_user_id_or_none
+                )
                 query = query.where(
-                    folders_to_projects.c.folder_id == f"{folder_id}"
-                    if folder_id
-                    else folders_to_projects.c.folder_id.is_(None)
+                    (projects.c.prj_owner == personal_workspace_user_id_or_none)
+                    | sa.text(
+                        f"jsonb_exists_any(access_rights_subquery.access_rights, {assemble_array_groups(user_groups)})"
```

Comment:
Q: this is not the first time I see that. but actually why are you using the jsonb calls with these? these are now all proper columns right?

Reply:
discussed